packages/utils/src/route.js

```diff
@@ -4,24 +4,37 @@ import consola from 'consola'
 
 import { r } from './resolve'
 
+const getRouteChildrens = function (route) {
+  if (route.children.some(child => child.path === '')) {
+    return route.children
+  }
+  return [
+    // Add default child to render parent page
+    {
+      ...route,
+      children: undefined
+    },
+    ...route.children
+  ]
+}
+
 export const flatRoutes = function flatRoutes (router, fileName = '', routes = []) {
   router.forEach((r) => {
     if ([':', '*'].some(c => r.path.includes(c))) {
       return
     }
-    fileName = (fileName + r.path + '/').replace(/\/+/g, '/')
+    if (r.children) {
+      return flatRoutes(getRouteChildrens(r), fileName + r.path + '/', routes)
```

Comment:
nit: using template string here instead of string concatention
```suggestion
      return flatRoutes(getRouteChildrens(r), `${fileName}${r.path}/`, routes)
```